### PR TITLE
Added HTTP lesson in TypeScript

### PR DESCRIPTION
**Pull request by lechnerc77**
Referring to issue #17 I have added a TypeScript version for the HTTP Trigger lesson based on the .NET HTTP Trigger lesson. 
I also added Code Tours for the different sections of the lesson and the homework in the corresponding source code folders.

### CODE REVIEW

**Comment by marcduiker** [comment]
Thanks @lechnerc77,  I'll start the review this week!

File: lessons/prerequisites-ts.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
+# Prerequisites
```
[the remark this answers is not in the document]
Comment (lechnerc77) [reply lessons/prerequisites-ts.md:1]: Done - updated the descriptions. Not sure if I am happy with the layout, so please check

File: lessons/prerequisites-ts.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+## Frameworks & Tooling 🧰
+
+In order to complete the the lessons using TypeScript you need to install the following:
+
+|Prerequisite|Lessons|Description
+|-|-|-
+|[Node.js](https://nodejs.org/en/download/)|All| The Node.js runtime including the node package manager NPM.
```
[the remark this answers is not in the document]
Comment (lechnerc77) [reply lessons/prerequisites-ts.md:9]: I added a remark with respect to LTS version. As LTS versions change every year I wouldn't pick a specific version.

File: lessons/http-ts.md [hunk 3]
```diff
@@ -0,0 +1,252 @@
+# HTTP Trigger
```
[the remark this answers is not in the document]
Comment (lechnerc77) [reply lessons/http-ts.md:1]: Done - I also updated the header in the .NET file

File: lessons/http-ts.md [hunk 4]
```diff
@@ -0,0 +1,252 @@
...
+    > ❔ **Question** - Is the outcome of the POST as expected?
+    > ❔ **Question** - What is the response when you use an empty `name` property?
+
+## 4. Adding a new function for POST requests
+
+In contrast to C# the typing of the request parameter of the function is restricted to `string`, `HttpRequest` and `buffer`, so we cannot have a type-safety on this level of the function. Nevertheless we can make use of implicitly assigning the JSON body of the request to an interface which gives us consistent typing at design time.
+
+### Steps
+
+1. Copy & paste the folder of the Azure Function from the exercise above and give it a new name e.g. CustomGreetingHttpTrigger.
```
[the remark this answers is not in the document]
Comment (lechnerc77) [reply lessons/http-ts.md:192]: Done

File: lessons/http-ts.md [hunk 5]
```diff
@@ -0,0 +1,252 @@
...
+|2|[Changing the template for GET requests](#2-changing-the-template-for-get-requests)
+|3|[Changing the template for POST requests](#3-changing-the-template-for-post-requests)
+|4|[Adding a new function for POST requests](#4-adding-a-new-function-for-post-requests)
+|5|[Homework](#5-homework)
+|6|[More info](#6-more-info)
+
+> 📝 **Tip** - If you're stuck at any point you can have a look at the [source code](../src/typescript/AzureFunctions.Http) in this repository.
+
+---
+
```
[the remark this answers is not in the document]
Comment (lechnerc77) [reply lessons/http-ts.md:23]: Done

File: lessons/http-ts.md [hunk 6]
```diff
@@ -0,0 +1,252 @@
...
+## 5. Homework
+
+Ready to get hands-on? Checkout the [homework assignment for this lesson](../homework/http_resume-api-ts.md).
+
+## 6. More info
+
+- For more info about the HTTP Trigger have a look at the official [Azure Functions HTTP Trigger](https://docs.microsoft.com/en-us/azure/azure-functions/functions-bindings-http-webhook-trigger?tabs=javascript) documentation.
+
+- A brief overview [video](https://youtu.be/Wbw6MS5VoDo) by Gwyneth Pena
```
[the remark this answers is not in the document]
Comment (lechnerc77) [reply lessons/http-ts.md:249]: Removed it

**Review by marcduiker — CHANGES_REQUESTED** [review]
This PR also contains the work from Stacy and myself on the Configuration lesson. Perhaps you need to rebase your branch on main on the original repo?

In addition the CodeTour files are not discovered in VSCode. I can open them manually but then the steps don't navigate to the right files (I do see the comments themselves).

**Comment by marcduiker** [comment]
Even when I add a new CodeTour it does not work, very weird. I'll try it on the main branch in my repo for a .NET project.

**Comment by lechnerc77** [comment]
Concerning the CodeTour => the files currently reference the main branch (in the JSON file of the code tour). If you change that to "http_ts" the code tour should work

For the git "chaos" - I have to check (especially if my git skills are sufficient to solve it

**Comment by marcduiker** [comment]
I pushed a small tour file in the `main` branch. I think the tour files all need to be in a `.tour` folder in the root. You can have subfolders inside that folder. https://github.com/microsoft/codetour#tour-files

**Comment by lechnerc77** [comment]
>Let me know, we can go on a call and fix this together if you'd like.

Help would be highly appreciated here - no idea what went sideways here.


> I pushed a small tour file in the `main` branch. I think the tour files all need to be in a `.tour` folder in the root. You can have subfolders inside that folder. https://github.com/microsoft/codetour#tour-files

I check the .tour file options and adjust accordingly

**Comment by lechnerc77** [comment]
Messed up git :-)

**Pull request closed without merging**